src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/UntypedObject.cs

```diff
@@ -45,4 +51,94 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
             yield return new[] { TexlStrings.IndexArg1, TexlStrings.IndexArg2 };
         }
     }
+
+    internal abstract class UntypedOrJSONConversionFunction : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        public override bool HasTypeArgs => true;
+
+        public override bool ArgIsType(int argIndex)
+        {
+            return argIndex == 1;
+        }
+
+        private static readonly ISet<DType> UnsupportedJSONTypes = new HashSet<DType> { DType.Color, DType.Void, DType.Time, DType.ObjNull };
```

Comment:
>UnsupportedJSONTypes [](http://example.com/codeflow?start=44&length=20)

Instead of having a list of unsupported types, it may be better to have a list of supported ones, otherwise we may add a type in the future, not add the proper support, and it this function will work incorrectly.